test/WebJobs.Script.Tests/Managment/InstanceManagerTests.cs

```diff
@@ -46,24 +65,82 @@ public class InstanceManagerTests
+
+            logs = _loggerProvider.GetAllLogMessages().Select(p => p.FormattedMessage).ToArray();
+            Assert.Collection(logs,
+                p => Assert.StartsWith("Assign called while host is not in placeholder mode", p));
         }
 
         [Fact]
         public void StartAssignment_ReturnsFalse_WhenNotInStandbyMode()
         {
-            var loggerFactory = MockNullLogerFactory.CreateLoggerFactory();
-            var settingsManager = new ScriptSettingsManager();
-            var instanceManager = new InstanceManager(settingsManager, null, loggerFactory, null);
-
             Assert.False(WebScriptHostManager.InStandbyMode);
 
             var context = new HostAssignmentContext();
-            bool result = instanceManager.StartAssignment(context);
+            bool result = _instanceManager.StartAssignment(context);
             Assert.False(result);
         }
+
+        [Fact]
+        public async Task ValidateContext_InvalidZipUrl_ReturnsError()
+        {
+            var environment = new Dictionary<string, string>()
+            {
+                { EnvironmentSettingNames.AzureWebsiteZipDeployment, "http://invalid.com/invalid/dne" }
+            };
+            var assignmentContext = new HostAssignmentContext
+            {
+                SiteId = 1234,
+                SiteName = "TestSite",
+                Environment = environment
+            };
+
+            string error = await _instanceManager.ValidateContext(assignmentContext);
+            Assert.Equal("Invalid zip url specified (StatusCode: NotFound)", error);
+
+            var logs = _loggerProvider.GetAllLogMessages().Select(p => p.FormattedMessage).ToArray();
+            Assert.Collection(logs,
+                p => Assert.StartsWith("Validating host assignment context (SiteId: 1234, SiteName: 'TestSite')", p),
+                p => Assert.StartsWith("Invalid zip url specified (StatusCode: NotFound)", p));
+        }
+
+        [Fact]
+        public async Task ValidateContext_Succeeds()
+        {
+            var environment = new Dictionary<string, string>()
+            {
+                { EnvironmentSettingNames.AzureWebsiteZipDeployment, "http://microsoft.com" }
+            };
+            var assignmentContext = new HostAssignmentContext
+            {
+                SiteId = 1234,
+                SiteName = "TestSite",
+                Environment = environment
+            };
+
+            string error = await _instanceManager.ValidateContext(assignmentContext);
+            Assert.Null(error);
+
+            var logs = _loggerProvider.GetAllLogMessages().Select(p => p.FormattedMessage).ToArray();
+            Assert.Collection(logs,
+                p => Assert.StartsWith("Validating host assignment context (SiteId: 1234, SiteName: 'TestSite')", p));
+        }
+
+        public void Dispose()
+        {
+            Environment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsitePlaceholderMode, null);
```

Comment:
These tests were missing env cleanup. We have to be super careful with that, because if we miss it the process env gets polluted causing test instability that is hard to diagnose